docs/defi/flow-yield-vaults/scheduling.md

```diff
@@ -0,0 +1,410 @@
+checkRatio()
+executeIfNeeded()
+  ↓
+// Reschedule next execution
+scheduleNextRebalance()
+  ↓
+FlowTransactionScheduler.schedule(
+    functionCall: "rebalance()",
+    executeAt: currentTime + 60 seconds
+)
+```
+
+**Perpetual Loop**:
+```
+Execute → Rebalance → Schedule Next → Wait 60s → Execute → ...
+```
+
+This creates an infinite loop where each rebalance execution schedules the next one, requiring no external coordination.
+
+### Atomic Registration
+
+Vault creation and scheduling registration happen atomically to prevent orphaned vaults:
+
+```cadence
+transaction createVault() {
+    prepare(signer: AuthAccount) {
+        // Create all components
+        let vault <- createYieldVault(...)
+        let autoBalancer <- createAutoBalancer(...)
+        let position <- createPosition(...)
+
+        // Register (all steps must succeed)
+        registerInRegistry(autoBalancer)     // Step 1
+        scheduleFirstRebalance(autoBalancer) // Step 2
+        linkComponents(...)                   // Step 3
+
+        // If ANY step fails → entire transaction reverts
+        // No partial vaults created
+    }
+}
+```
+
+**Atomicity guarantee**: Either vault is fully created with working schedule, OR transaction fails and nothing is created.
+
+## SchedulerRegistry
+
+The SchedulerRegistry maintains a global record of all active vaults and their scheduling state.
+
+### Registry Structure
+
+```cadence
+pub contract FlowYieldVaultsSchedulerRegistry {
+    // Maps vault ID → scheduling info
+    access(contract) var registry: {UInt64: ScheduleInfo}
+
+    pub struct ScheduleInfo {
+        pub let vaultID: UInt64
+        pub let autoBalancerCap: Capability<&AutoBalancer>
+        pub let nextScheduledTime: UFix64
+        pub let status: ScheduleStatus  // Active, Pending, Stuck
+    }
+
+    pub enum ScheduleStatus: UInt8 {
+        pub case Active    // Scheduling working normally
+        pub case Pending   // Awaiting schedule
+        pub case Stuck     // Failed to reschedule
+    }
+}
+```
+
+### Registration Lifecycle
+
+**On vault creation:**
+```cadence
+registry.register(
+    vaultID: 42,
+    autoBalancerCap: capability,
+    status: ScheduleStatus.Pending
+)
+```
+
+**After first successful schedule:**
+```cadence
+registry.updateStatus(
+    vaultID: 42,
+    status: ScheduleStatus.Active,
+    nextScheduledTime: currentTime + 60
+)
+```
+
+**If schedule fails:**
+```cadence
+registry.updateStatus(
+    vaultID: 42,
+    status: ScheduleStatus.Stuck
+)
+// Supervisor will attempt recovery
+```
+
+**On vault liquidation:**
+```cadence
+registry.unregister(vaultID: 42)
+// Vault removed from tracking
+```
+
+## Supervisor Recovery System
+
+The Supervisor handles vaults that become stuck or fail to self-schedule.
+
+### What Can Go Wrong?
+
+Despite atomicity guarantees, vaults can become stuck for several reasons:
+
+1. **Transaction failure** during reschedule due to gas issues or network congestion
```

Comment:
It's not super clear who pays for gas with the rescheduling or if you need to fund an account that sponsors the txns indefinitely